common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java

```diff
@@ -99,9 +99,8 @@ public static URL appendPathAndQueryToURL(@NonNull final URL urlToAppend,
         }
 
         builder.setPathSegments(pathSegments);
-        if (queryParam != null && !queryParam.isEmpty()) {
-            builder.setQuery(queryParam);
-        }
+        builder.addParametersIfAbsent(queryParam);
```

Comment:
now that we remove the check do, we have tests for null or empty map?

Reply:
added tests in CommonURIBuilderTests.